isort/main.py

```diff
@@ -1087,7 +1087,9 @@ def main(argv: Optional[Sequence[str]] = None, stdin: Optional[TextIOWrapper] =
         return
     if "settings_path" not in arguments:
         arguments["settings_path"] = (
-            os.path.abspath(file_names[0] if file_names else ".") or os.getcwd()
+            arguments.get("filename", None) or os.getcwd()
```

Comment:
```suggestion
            os.path.abspath(arguments.get("filename", None) or os.getcwd())
```
Should `--filename` argument be wrapped with `abspath` as well?

Reply:
I think that's confusing because then it depends on the working dir again.  The working dir is not as easy to get right for editors; it might be the topmost open folder, or the first folder with .git in it.  And then we need to construct a relative filename of course to set it as the argument.